tests/integration/sharding_tests/helpers.py

```diff
@@ -62,44 +64,105 @@ async def deploy_cluster_components(
         if channel
         else {"mongodb-image": METADATA["resources"]["mongodb-image"]["upstream-source"]}
     )
-    await ops_test.model.deploy(
-        my_charm,
-        resources=resources,
-        num_units=num_units_cluster_config[CONFIG_SERVER_APP_NAME],
-        config={"role": "config-server"},
-        application_name=CONFIG_SERVER_APP_NAME,
-        channel=channel,
-        series="jammy",
-        trust=True,
-    )
-    await ops_test.model.deploy(
-        my_charm,
-        resources=resources,
-        num_units=num_units_cluster_config[SHARD_ONE_APP_NAME],
-        config={"role": "shard"},
-        application_name=SHARD_ONE_APP_NAME,
-        channel=channel,
-        series="jammy",
-        trust=True,
-    )
-    await ops_test.model.deploy(
-        my_charm,
-        resources=resources,
-        num_units=num_units_cluster_config[SHARD_TWO_APP_NAME],
-        config={"role": "shard"},
-        application_name=SHARD_TWO_APP_NAME,
-        channel=channel,
-        series="jammy",
-        trust=True,
-    )
-
-    await ops_test.model.wait_for_idle(
-        apps=CLUSTER_COMPONENTS,
-        idle_period=20,
-        timeout=TIMEOUT,
-        raise_on_blocked=False,
-        raise_on_error=False,
-    )
+    if channel is None:
```

Comment:
why is the code so duplicated?

Reply:
It's not duplicated, but there's a bug currently (fixed by this release) that makes the deployment fail to init replicaset from time to time so in order to prevent that we deploy with one and then scale up.
This will be reverted right after that release with the new code that fixes this bug.